### ISSUE TITLE
Wrong timestamp for events (all shifted by an hour)

### ISSUE THREAD

**Issue by zoltanf** [ak--47/mpMigrate#10]
Both source and destination projects are in the same time zone (source has US and destination EU data residency), but all timestamps are shifted by one hour. 
Guessing here that the export does use local time zone and the import uses UTC, hence the time difference; or it could be something else. 

Source project events for a profile:
![events-timestamps-profile-mixpanel](https://user-images.githubusercontent.com/1464667/225581533-d1be7063-e8fe-458e-aae7-ec4e99c546a6.png)

Destination project events for a profile:
![events-timestamps-profile-eu-mixpanel](https://user-images.githubusercontent.com/1464667/225581530-0ec839e8-2d3c-4458-b566-3a686883a74a.png)

Would it be possible to apply a transformation that would compensate for this while moving the events?
What about people properties that use date/time?

Thank you, your help is greatly appreciated.

**Comment by ak--47** [comment]
@zoltanf acknowledged and can reproduce! 👍

the short answer is that mixpanel has changed how `time` (event time) is stored and parsed... projects created before Feb 2023 did not store `time` in UTC, so when we `/export` data on older projects, we're actually getting _local_ time (not UTC time)

projected created after Feb 2023 _do the right thing_ (everything stored in UTC) time, but for your particular use-case this is not helpful 🤪

i will be adding a `timezoneOffset` param in the configuration to adjust for this. 

note this has no bearing on people/user props. "date" props sent to user profiles are _not_ transformed: https://help.mixpanel.com/hc/en-us/articles/115004547063-Properties-Supported-Data-Types#date

will update you when it's shipped.

thanks for reporting!

**Comment by ak--47** [comment]
cc @alon-mixpanel

**Comment by ak--47** [comment]
@zoltanf 

this is fixed in `v1.2.0`
https://github.com/ak--47/mpMigrate/pull/11

you'll specify a `timeOffset` in your `options` like so:

```javascript
 "options": {
    "timeOffset": 8, //add or subtract this number of hours from every event before sending
    "shouldGenerateSummary": true,
    "shouldCopyEvents": true,
    "shouldCopyProfiles": true,
    "shouldCopySchema": true,
    "shouldCopyEntities": false,
    "skipPrompt": true
  }
```

i will send you a follow-up email tomorrow - but if you want to test it now, just make sure you're using `mp-migrate@latest`

**Comment by zoltanf** [comment]
Thank you @ak--47 
Will give it a spin today.
